### PR TITLE
generate_virtual_pop function finshed, please review

### LINKED ISSUE #21: Finish generate_virtual_pop function

### LINKED ISSUE #24: Finish stat_summary function

### LINKED ISSUE #27: Write stat_summary test function

### LINKED ISSUE #39: Add integer input testing for generate_virtual_pop

### LINKED ISSUE #40: Fix inconsistent code format
For both generate_virtual_pop and stat_summary

### LINKED ISSUE #41: Add comments to the code
Per Lise's advice. For both functions

### PR DESCRIPTION
**Pull request by tguo9**
Address  #21 
Please test and review

### CODE REVIEW

File: R/generate_virtual_pop.R [hunk 1]
```diff
@@ -13,6 +13,14 @@
...
 #'generate_virtual_pop(100, "n_accidents_per_hr", rpois, 2)
-generate_virtual_pop <- function(N, var_name, dist, ... ){
+#'
+#' @importFrom rlang :=
+#' @export
+generate_virtual_pop <- function(N, var_name, dist, ...) {
+  distribution <- match.fun(dist)
+
+  var_name <- dplyr::enquo(var_name)
+  pop <- dplyr::tibble({{var_name}} := distribution(n = N, ...))
```
Comment (lisebraaten) [inline R/generate_virtual_pop.R]: corrected to dplyr::tibble

**Review by lisebraaten — APPROVED** [review]

**Review by lisebraaten — CHANGES_REQUESTED** [review]
Include exception handling in your function, raise error messages for incorrect inputs

**Comment by tguo9** [comment]
Nice catch! I updated it, addressed in 3053cfa

File: R/generate_virtual_pop.R [hunk 2]
```diff
@@ -13,6 +13,23 @@
 #'@examples
 #'generate_virtual_pop(100, "height", rnorm, 0, 1)
 #'generate_virtual_pop(100, "n_accidents_per_hr", rpois, 2)
+#'
+#' @importFrom rlang :=
+#' @export
 generate_virtual_pop <- function(N, var_name, dist, ... ){
```
Comment (lisebraaten) [inline R/generate_virtual_pop.R]: do you think N input needs to be tested? I tried it with decimal values and it rounds down and takes that many samples... what do you think?

File: R/generate_virtual_pop.R [hunk 3]
```diff
@@ -13,6 +13,23 @@
...
+  if (class(try(match.fun(dist), silent=TRUE)) == "try-error") {
+    stop('This is not a vaild function, please check https://cran.r-project.org/web/views/Distributions.html for avaliable distribution functions')
```
Comment (lisebraaten) [inline R/generate_virtual_pop.R:31]: good addition of link

File: R/stat_summary.R [hunk 4]
```diff
@@ -11,6 +11,29 @@
...
+  pop <- dplyr::tibble('data' := c('population', 'samples'))
+
+  for (i in (1:length(parameter))) {
+    if (class(try(match.fun(parameter[i]), silent = TRUE)) == "try-error") {
+
+      stop('This is not a vaild function')
+
+    } else {
+      para_func <- match.fun(parameter[i])
+    }
```
Comment (lisebraaten) [inline R/stat_summary.R]:
```suggestion
  for (i in (1:length(parameter))) {
    if (class(try(match.fun(parameter[i]), silent = TRUE)) == "try-error") {
      stop('This is not a vaild function')
    } else {
      para_func <- match.fun(parameter[i])
    }
```

spacing to match rest of code

File: R/generate_virtual_pop.R (hunk 2 above)
Comment (lisebraaten) [reply R/generate_virtual_pop.R]: also comments through out code? eg. what  exceptions are checking for

**Review by lisebraaten — CHANGES_REQUESTED** [review]
hey Tao! Tests look good, minor few comments I left... let me know what you think. Thanks!

File: R/generate_virtual_pop.R (hunk 2 above)
Comment (tguo9) [reply R/generate_virtual_pop.R]: > do you think N input needs to be tested? I tried it with decimal values and it rounds down and takes that many samples... what do you think?

I think you are correct! I opened an issue addressing this. #39

File: R/stat_summary.R (hunk 4 above)
Comment (tguo9) [reply R/stat_summary.R]: Great point! I will check both of my functions. Addressed #40

File: R/generate_virtual_pop.R (hunk 2 above)
Comment (tguo9) [reply R/generate_virtual_pop.R]: > do you think N input needs to be tested? I tried it with decimal values and it rounds down and takes that many samples... what do you think?

That's definitely a good idea. Addressed #41

**Review by lisebraaten — APPROVED** [review]
Thank Tao for making changes, will merge and test output with my function I am working on as well

**Pull request merged**